add NeoTreeFloatTitle highlight group

hbiel opened the pull request:
This fixes the issue described in https://github.com/nvim-neo-tree/neo-tree.nvim/discussions/35. I cleaned everything up and updated the documentation accordingly.

If there's something missing please let me know!

Best regards!

cseickel commented:
This looks good, I'll just try it out later and I'll merge if there are no issues.

cseickel commented:
The problem with this is that I don't think the TitleBar highlight group looks good with a normal border. It tried using the FloatBorder group for the title and that's better but still not great. I think the right thing to do is to add another highlight group which uses the same background as float border but with the Normal foreground.

cseickel commented:
I just added a commit to show you what I mean, what do you think about that? Docs would still need to be updated.

hbiel commented:
I think the 'NC' style actually looks better on my initial commit. Although 'rounded' looks better on yours.
I'm using kanagawa.nvim as my theme so you might be getting different results.
Here is a comparison for both cases on my current setup:

'NC' style my commit: ![6d46faf0eb43b46ead7c04ff54a3e39c10369d05_NC_border](https://user-images.githubusercontent.com/18981810/156817537-4e9c9105-76e3-40ab-95ea-4a9a601b1d36.jpg)
'NC' style your commit: ![093a1fe20abd7080b3b5a4e538327a48ba953ff7_NC_border](https://user-images.githubusercontent.com/18981810/156817588-f7283ca3-c642-45f4-aa75-381a46968370.jpg)

'rounded' style my commit: ![6d46faf0eb43b46ead7c04ff54a3e39c10369d05_rounded_border](https://user-images.githubusercontent.com/18981810/156817697-592921d1-382e-49c7-851a-e9d0ee217f11.jpg)
'rounded' style your commit: ![093a1fe20abd7080b3b5a4e538327a48ba953ff7_rounded_border](https://user-images.githubusercontent.com/18981810/156817731-d456d42b-24b1-47e3-99d2-37ecd9b15141.jpg)

I added another commit in which different highlight groups are used as each case actually needs different highlighting.
In 'NC' style the title border is drawn by the background highlighting of blank characters.
In the other styles the border is drawn by actual glyphs and as such the foreground highlighting is used.

It now looks like this for me:
'NC' style uses the established TitleBar highlight group and looks like this: ![NC style](https://user-images.githubusercontent.com/18981810/156823658-57c100f8-598e-4505-a5f8-2c527cbd43b1.jpg)
'rounded' style uses the FloatBorder highlight group and looks like this (just like telescope does for me): ![rounded style](https://user-images.githubusercontent.com/18981810/156823694-a7bae194-2744-4ebb-ad4a-917f6b21aefc.jpg)
 
The additional highlight group isn't needed anymore as far as i can tell so i removed it.

Does this also look like this on your end or are you seeing different results?

cseickel commented:
I didn't even check the NC style because I didn't expect it to be affected. If you back out the changes you made to that branch then it should work out correctly.

I'd like to keep the added highlight group because I think it's something that others would want to be able to change. I know I personally don't like the way it looks when the title text is the same color as the border.  We can, however, make the default for that new group just be linked to the FloatBorder group so the default look is still what you are looking for.

cseickel commented on lua/neo-tree/ui/popups.lua:
```diff
@@ -44,12 +44,11 @@ M.popup_options = function(title, min_width, override_options)
 
   if popup_border_style == "NC" then
     local blank = NuiText(" ", highlights.TITLE_BAR)
-    local text = NuiText(" " .. title .. " ", highlights.TITLE_BAR)
```
This should not have been removed

cseickel commented on lua/neo-tree/ui/popups.lua:
```diff
@@ -44,12 +44,11 @@ M.popup_options = function(title, min_width, override_options)
       style = { "▕", blank, "▏", "▏", " ", "▔", " ", "▕" },
       highlight = highlights.FLOAT_BORDER,
       text = {
-        top = text,
```
This also should be reverted.

hbiel commented:
> I'd like to keep the added highlight group because I think it's something that others would want to be able to change. I know I personally don't like the way it looks when the title text is the same color as the border. We can, however, make the default for that new group just be linked to the FloatBorder group so the default look is still what you are looking for.

Yeah, that's nice to have. I have restored this in the latest commit and kept the defaults you set initially as they seem perfectly fine to me. I'm more than happy to be able to set my own colors in my personal configuration.

I have also added documentation for the new highlight group and have grouped the three groups together.

cseickel reviewed (approved):
Perfect, thanks!
